### DISCUSSION TITLE
code-server startup and load very slow

### DISCUSSION THREAD

**Discussion by VeAmas** [coder/code-server#6596] [Q&A] [answered]
Hello! I successfully launched code-server in Ubuntu22.04 (virtual-machine), but the startup process was very slow (took about 2 minutes), and the resource loading on the page was also very slow.
```
amas@amas-virtual-machine:~/Git/code-server$ yarn watch
yarn run v1.22.21
$ VSCODE_DEV=1 VSCODE_IPC_HOOK_CLI= NODE_OPTIONS='--max_old_space_size=32384 --trace-warnings' ts-node ./ci/dev/watch.ts
```
It doesn't seem to be a memory issue. How can I speed up the page loading?

```
amas@amas-virtual-machine:~/.config/code-server$ free
               total        used        free      shared  buff/cache   available
Mem:        30613084    13950972     2627312      384536    14034800    15829660
Swap:        2097148           0     2097148

```

![image](https://github.com/coder/code-server/assets/19408439/668dd82a-96e3-41ec-90bd-212996c2c55d)

**Comment by VeAmas** [comment]
It looks like code-server is reaching 100% usage of the CPU. Does this mean that my CPU performance is a bottleneck?
![image](https://github.com/coder/code-server/assets/19408439/eda3eaf7-8d59-4e37-aa8c-7caba36d4262)

**Comment by code-asher** [comment]
Development can be pretty resource-intensive, but CPU usage should only go up while it is compiling the first time.  Do you need dev mode or can you run an optimized build instead?

**Reply by code-asher** [reply, marked as answer]
Unfortunately dev mode loads thousands of files in sequence due to how Microsoft architectured it so the slow load is unavoidable.  A complete refactor upstream would be ideal.

**Reply by VeAmas** [reply]
It seems that  there's nothing I can do about it.  Thanks a lot
